str/trtools/merge_str_prep.py

```diff
@@ -0,0 +1,143 @@
+        remote_tmpdir=remote_tmpdir(),
+    )
+    b = hb.Batch(backend=backend, default_image=os.getenv('DRIVER_IMAGE'))
+
+    external_id_to_cpg_id: dict[str, str] = SampleApi().get_sample_id_map_by_external(
+        dataset, list(external_wgs_ids)
+    )
+
+    # Working with CRAM files requires the reference fasta
+    ref = b.read_input_group(
+        **dict(
+            base=REF_FASTA,
+            fai=REF_FASTA + '.fai',
+            dict=REF_FASTA.replace('.fasta', '').replace('.fna', '').replace('.fa', '')
+            + '.dict',
+        )
+    )
+
+    input_vcf_dict = {}
+
+    if caller == 'eh':
+        for id in list(external_id_to_cpg_id.values()):
+            input_vcf_dict[id] = input_dir + '/' + id + '_EH.vcf'
+    elif caller == 'gangstr':
+        for id in list(external_id_to_cpg_id.values()):
+            input_vcf_dict[id] = input_dir + '/' + id + '_gangstr.vcf'
+    else:
+        raise Exception('Invalid caller')
+
+    for id in list(input_vcf_dict.keys()):
+
+        bcftools_job = b.new_job(name=f'{id} {caller} Files prep')
+        bcftools_job.image(BCFTOOLS_IMAGE)
+        bcftools_job.storage('20G')
+        bcftools_job.cpu(8)
+
+        vcf_input = b.read_input(input_vcf_dict[id])
+
+        if caller == 'eh':
+            bcftools_job.declare_resource_group(
+                vcf_sorted={
+                    'vcf.gz': '{root}.vcf.gz',
+                    'reheader.vcf.gz': '{root}.reheader.vcf.gz',
+                    'vcf.gz.tbi': '{root}.reheader.vcf.gz.tbi',
+                }
+            )
+            bcftools_job.command(
+                f"""
+
+                bgzip -c {vcf_input} > {bcftools_job.vcf_sorted['vcf.gz']}
+            
+                bcftools reheader -f {ref.fai} -o {bcftools_job.vcf_sorted['reheader.vcf.gz']} {bcftools_job.vcf_sorted['vcf.gz']} 
+
+                tabix -f -p vcf {bcftools_job.vcf_sorted['reheader.vcf.gz']} > {bcftools_job.vcf_sorted['vcf.gz.tbi']}
+            
+                """
+            )
+            # Output writing
+            output_path_eh = output_path(f'{id}_eh')
+            b.write_output(
+                bcftools_job.vcf_sorted['reheader.vcf.gz'],
+                output_path_eh + '.reheader.vcf.gz',
+            )
+            b.write_output(
+                bcftools_job.vcf_sorted['vcf.gz.tbi'],
+                output_path_eh + '.reheader.vcf.gz.tbi',
+            )
+
+        else:
+            bcftools_job.declare_resource_group(
+                vcf_sorted={
+                    'vcf.gz': '{root}.vcf.gz',
+                    'vcf.gz.tbi': '{root}.reheader.vcf.gz.tbi',
+                }
+            )
+            bcftools_job.command(
+                f"""
+
+                bcftools sort {vcf_input} | bgzip -c  > {bcftools_job.vcf_sorted['vcf.gz']}
+            
+                tabix -f -p vcf {bcftools_job.vcf_sorted['vcf.gz']} > {bcftools_job.vcf_sorted['vcf.gz.tbi']}
```

Comment:
```suggestion
                tabix -p vcf {bcftools_job.vcf_sorted['vcf.gz']}
```
I don't think tabix `.tbi` outputs can be directed, it's always written to `input_file.tbi`

Reply:
It seems to work with `cpg-tob-wgs-test` data?

Reply:
It'll write it to that location, but tabix as a tool doesn't let you direct output to a specific file - it always just creates a file called (input_filename) + 'tbi'. 

This change isn't going to change the created file, but it's just good to point out that when you specify the filename here it isn't doing anything.